File: src/Web/JavaScriptJS/patterns/mvvm.js
const assert = require("node:assert/strict");
/** Exposes presentation-ready observable state through a view model. */
function run() {
  const model = { first: "Ada", last: "Lovelace" };
  const viewModel = {
    get fullName() { return `${model.first} ${model.last}`; },
    rename(first, last) { model.first = first; model.last = last; },
  };
  assert.equal(viewModel.fullName, "Ada Lovelace");
  viewModel.rename("Grace", "Hopper");
  assert.equal(viewModel.fullName, "Grace Hopper");
  return "mvvm";
}
module.exports = { run };
if (require.main === module) run();

File: src/Scripting/Ruby/patterns/mvvm.rb
# frozen_string_literal: true

module RubyPatterns
  module MVVM
    module_function

    def run
      model = { first: 'Ada', last: 'Lovelace' }
      view_model = -> { { display_name: "#{model[:first]} #{model[:last]}" } }
      raise 'MVVM failed' unless view_model.call[:display_name] == 'Ada Lovelace'
      true
    end
  end
end

RubyPatterns::MVVM.run if $PROGRAM_NAME == __FILE__
